ihatemoney/currency_convertor.py

```diff
@@ -0,0 +1,41 @@
+import requests
+from diskcache import Cache
+
+
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class CurrencyConverter(object, metaclass=Singleton):
+    # Convert all to USD
+    api_url = "https://api.exchangerate-api.com/v4/latest/USD"
+    _internal_cache = Cache()
+
+    def __init__(self, expire=86400):
+        self.expire = expire
+
+    def set_rate(self):
+        if "rates" not in self._internal_cache:
+            self._internal_cache.set(
+                "rates", requests.get(self.api_url).json()["rates"], expire=self.expire
```

Comment:
The `parse_float` argument from [`json.loads`](https://docs.python.org/3/library/json.html#json.load) is even given in the examples.